Ricoh SP 210 SU and 211 drivers

abhisekp opened the pull request:
# Drivers for Ricoh 210 SU and 211

> PPD file

Printing works fine. But scanning does't work.

necessary129 commented:
The filter file is not included in the pull request.

abhisekp commented:
@necessary129 the PPD file is all that was included. I got them from the service center. And that's all that was required to install the driver. I don't have anything more.

necessary129 commented:
@abhisekp 😮 Can you share the link?

abhisekp commented:
@necessary129 Sorry! I didn't get you. I've shared everything I've got (*even the test page* :hushed: ). Link to where? I got it directly from Ricoh service center at my locality. 

I've seen the list of drivers they've. These are not shared publicly (*it's in Ricoh's private download page*) but one can visit the service center and get it from them directly.

There was a firmware driver for Ricoh SP 210 SU Printer (*which I didn't get*).

necessary129 commented:
@abhisekp Oh! I thought you downloaded it from Ricoh's website. Could you upload the `rastertolilo` file somewhere?

abhisekp commented:
@necessary129 I've updated the pull with the exact two files that I got from service center i.e. a `.deb` file and a `.rpm`. 

I extracted them to get the PPD files. I just saw there is a `rastertolilo` file too after extracting (*inside one of the directories*).

necessary129 commented:
Thank you. Thank is what I was looking for :)

aadityabagga commented:
Can confirm that after installing the `.deb` file I was able to print the test page on Ubuntu 16.04.

Thx guys @abhisekp and @necessary129 

:-)

zwpwjwtz commented:
Thank you very much, abhisekp.
I almost gave up when I found this commit. You save for me 40€!

grinya007 commented:
Guys, I've just found official Mac OS driver for SP 211 printer on official Ricoh website, it's either some complicated marketing strategy (say, they pretend that this cheap printer is for Windows only to force Mac users buy more expensive models), or just silly mistake of someone responsible for publishing of drivers on website. Either way, long story short: do google for "ricoh sp 221 driver" -> follow first link -> choose Mac OS -> install driver from downloaded dmg -> choose "RICOH SP 221" driver upon installation (important here is not to choose drivers for ...N, ...w, ...Nw and such printers, all Ricoh printers with network connectivity are PCL printers, whereas this one is GDI, they call it DDST whatever that means). You'll find working driver for Linux there as well. And one more funny fact - it turns out that SP 221 model just doesn't exist at all =) really, try order it somewhere. Hope that helps, enjoy!

MacHoliday commented:
I am trying to install the Ricoh SP 211 Printer on my raspberry. The filter file "rastertolilo" is compiled for x86 architecture. So is it possible to get the source files or compiled for arm/raspberry?

grinya007 commented:
@MacHoliday I don't think so, I've spent quite some time googling exactly same question, found nothing and gave up. But, if you don't mind paying ~$15 and waiting ~20 sec before each job starts actual printing: have a look at exagear by eltechs.com which allows to run x86 binaries on arm arch, you'll have 3 days of trial to make decision whether it fits your needs. I did so, it really works smoothly (yet slowly of course). I tried to run only Ricoh's filter binary under x86 emulation while CUPS itself run natively but it didn't work for me. So, I ended up with minus $15 in my pocket, CUPS daemon and Google Cloud Print connector running under x86 emulation (which feels like shooting sparrows with a massive cannon), but I'm still happy with it =)

thewebdev commented:
Signed in to thank @grinya007 - your tip worked for Ricoh SP 210 SU too on my macOS 10.12!! (And I didn't even have to download the drivers from Ricoh site as macOS has it built-in).

I installed my printer on Mac by selecting the Ricoh SP 221S, 1.01 "printer software" and my Ricoh SP 210SU now partially works with my Mac OS too - it prints fine, but the scanner doesn't work.

erotavlas85 commented:
Hi,
I have a similar problem. I successfully installed the driver for sp211su on ubuntu 18.04 64 bit. However, the scanner still does not work. Is it only possible to use it via scan to USB?

abhisekp commented:
@erotavlas85  yes. The scanner has to be via USB. It's an OK compensation to make the device work with linux. :smile_cat:

LAfricain commented:
Hello, 
Is the `rastertolilo` opensource?

LAfricain commented:
I just created a new deb package with all the ppd of all the branches. 
https://launchpad.net/~lafricain79/+archive/ubuntu/lafricainenvrac

abhisekp commented:
https://github.com/madlynx/ricoh-sp100/pull/64#issuecomment-537523121

No idea!

yoabhi commented:
> > Hello,
> > Is the rastertolilo opensource?
> > [#64 (comment)](https://github.com/madlynx/ricoh-sp100/pull/64#issuecomment-537523121)
> 
> No idea!
I have followed all steps. but CUPS showing stopped 
"Filter failed"

yoabhi commented:
@abhisekp  Hello Please guide me how to install Sp 210 Su printer driver in Raspberry pi Zero W ?

abhisekp commented:
> @abhisekp Hello Please guide me how to install Sp 210 Su printer driver in Raspberry pi Zero W ?

I'm not sure how to do that. Someone else who has worked with IOT can pitch in.

yoabhi commented:
> The rasterlilo file can be used on a Raspberry Pi using the qemu-i386 package
> For that you need all shared libraries rastertolilo uses
> Execute these command on a x86 machine (Laptop/PC) which has the driver installed and working.
> 
> `cd ~`
> `mkdir ricoh`
> `cd ./ricoh`
> `cp /usr/lib/cups/filter/rastertolilo .`
> `mkdir qemu-i386`
> `ldd rastertolilo | gawk '{print $3}' | xargs -n 1 cp -t ./qemu-i386`
> `cp /lib/ld-linux.so.2 .`
> `tar -cvzf libs.tar.gz ld-linux.so.2 qemu-i386/`
> `scp libs.tar.gz pi@YOUR-RPI-IP:~`
> 
> Once this is done ssh your RPI install drive using the DEB file and execute the following commands.
> `cd ~`
> `tar -xzvf libs.tar.gz`
> `sudo cp ./ld-linux.so.2 /lib/`
> `sudo cp -r qemu-i386/ /lib/`
> `sudo cp /usr/lib/cups/filter/rastertolilo /usr/lib/cups/filter/rastertolilo.x86`
> `sudo cp /dev/null /usr/lib/cups/filter/rastertolilo`
> 
> Open the /usr/lib/cups/filter/rastertolilo file in your favourite text editor
> and write following lines to the file .
> 
> #!/bin/sh
> export LD_LIBRARY_PATH=/lib/qemu-i386
> exec qemu-i386 /usr/lib/cups/filter/rastertolilo.x86 "$@"
> 
> That's it you are done!!
> Try executing `/usr/lib/cups/filter/rastertolilo` and confirm it dosen't throw an "Execv error"
> Your printer should work with your RPI now.
> 
> Further follow the guide mentioned below to setup Google Cloud print service on RPI, provided you have properly setup CUPS on your RPI:
> https://github.com/google/cloud-print-connector/wiki/Installing-on-Raspberry-Pi-Raspbian-Jessie

Have you tried on Raspberry pi zero W?

mynemons commented:
Raspberry pi 3, nothing. Why? 😓


pi@raspberrypi:~ $ sudo su
root@raspberrypi:/home/pi# /usr/lib/cups/filter/rastertolilo
/usr/lib/cups/filter/rastertolilo: 3: exec: qemu-i386: not found
root@raspberrypi:/home/pi#



#!/bin/sh
export LD_LIBRARY_PATH=/lib/qemu-i386
exec qemu-i386 /usr/lib/cups/filter/rastertolilo.x86 "$@"

mynemons commented:
Please can anyone help me setting qemu emulator for using rastertolilo driver in raspberry? I only have X64 machine or x86/x64 cpu
